src/DevBetterWeb.Infrastructure/Services/DiscordWebhookService.cs

```diff
@@ -0,0 +1,46 @@
+﻿using DevBetterWeb.Infrastructure.DiscordWebhooks;
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using DevBetterWeb.Core.Interfaces;
+using DevBetterWeb.Infrastructure.Interfaces;
+
+namespace DevBetterWeb.Infrastructure.Services;
+
+public class DiscordWebhookService : IDiscordWebhookService
+{
+	private readonly HttpClient _httpClient = new();
+	private readonly IAppLogger<DiscordWebhookService> _logger;
+
+	public DiscordWebhookService(IAppLogger<DiscordWebhookService> logger)
+	{
+		_logger = logger;
+	}
+
+	public Task<HttpResponseMessage> SendAsync(ulong id, string token, string contentBody, string? username = null, string? avatarUrl = null, bool isTTS = false, IEnumerable<Embed>? embeds = null)
+	{
+		var webhookUrl = $"https://discordapp.com/api/webhooks/{id}/{token}";
+
+		return SendAsync(webhookUrl, contentBody, username, avatarUrl, isTTS, embeds);
+	}
+
+	public async Task<HttpResponseMessage> SendAsync(string webhookUrl, string contentBody, string? username = null, string? avatarUrl = null, bool isTTS = false, IEnumerable<Embed>? embeds = null)
+	{
+		var request = new WebHookRequest(contentBody, username, avatarUrl, isTTS, embeds);
+		var json = JsonSerializer.Serialize(request);
+
+		using var content = new StringContent(json, Encoding.UTF8, "application/json");
+		try
+		{
+			return await _httpClient.PostAsync(webhookUrl, content);
+		}
```

Comment:
@ShadyNagy How are we handling the case where we are not getting a success here?

Reply:
And why are we returning the Response if we never do anything with it?

Reply:
> And why are we returning the Response if we never do anything with it?

I think it is the same as what you did in your code `return response;` ! ;) This also need to use using in `var content = new StringContent(JsonConvert.SerializeObject(this), Encoding.UTF8, "application/json");`
```csharp
public virtual async Task<HttpResponseMessage> Send()
	{
		var content = new StringContent(JsonConvert.SerializeObject(this), Encoding.UTF8, "application/json");
		try
		{
			var response = await _httpClient.PostAsync(_webhookUrl, content);
			var responseContent = await response.Content.ReadAsStringAsync();
			response.EnsureSuccessStatusCode();
			return response;
		}
		catch (Exception ex)
		{
			_logger.LogError(ex, "Error sending webhook", _webhookUrl, content);
			return new HttpResponseMessage();
		}
	}
```

Reply:
> @ShadyNagy How are we handling the case where we are not getting a success here?

EnsureSuccessStatusCode is added

Reply:
If we're not using the response we might as well not return it from the function. This way we'll reduce possible confusion in future readers.

And do you think we can actually log the response content if ever the request is not successful? It will help with diagnostics if ever there's a problem.